src/sw.ts

```diff
@@ -285,7 +301,15 @@ function isAggregateError (err: unknown): err is AggregateError {
 
 function isSwAssetRequest (event: FetchEvent): boolean {
   const isActualSwAsset = /^.+\/(?:ipfs-sw-).+$/.test(event.request.url)
-  return isActualSwAsset
+  // if path is not set, then it's a request for index.html which we should consider a sw asset
+  const url = new URL(event.request.url)
+  // but only if it's not a subdomain request (root index.html should not be returned for subdomains)
+  const isIndexHtmlRequest = url.pathname === '/' && !isSubdomainRequest(event)
+
+  // if request is for https://cdn.jsdelivr.net/* we should consider it a sw asset
+  const isJsDelivrRequest = /^https:\/\/cdn.jsdelivr.net\/.+$/.test(event.request.url)
```

Comment:
Tangentially related: what do you think about publishing the CSS as part of this app instead of relying on a third-party origin (jsdeliver).

With both Cloudflare and the SW Cache, should be performant enough.

Reply:
now that i'm at my laptop, I think i started to do that with the dynamic imports, but it only made the bundle larger.. now that index.ts and app.tsx is smaller, and we have a router, I could include tachyons and ipfs-css on only helper-ui, redirect-page, and config, and it will reduce burden for redirect-interstitial (ipfs first-hit without sw registered)

Reply:
fixed by [`20a9c6d` (#234)](https://github.com/ipfs-shipyard/service-worker-gateway/pull/234/commits/20a9c6d0f92a4dbf63d556d062b52826c1f2627f)

note that because the bundle is larger now (`-rw-r--r--  1 sgtpooki  13M Apr 30 08:08 ipfs-sw-138-66da405c07a0317cdbbf.css`), there is FOUC on routing determination.

I don't think we should include it in the bundle.. but also, I think we should because then we're not dependent upon a third party.. ¯\\\_(ツ)\_/¯

Reply:
@lidel could use eyes on this one here.